timesketch/frontend/src/components/Sigma/SigmaList.vue

```diff
@@ -127,14 +204,165 @@ export default {
+      // I am actually not sure what this is for, but it works.
+    },
+    parseSigma: function (rule_yaml) { // eslint-disable-line
+      this.parsing_issues = []
+      ApiClient.getSigmaRuleByText(rule_yaml)
         .then(response => {
           let SigmaRule = response.data.objects[0]
           this.parsed = SigmaRule
+          this.data_type_present = (SigmaRule['search_query'].includes("data_type") || SigmaRule['search_query'].includes("source_name"))
+
+        })
+        .catch(e => {
+          Snackbar.open({
+            message: 'Sigma rule parsing failed. See Browser console for more',
+            type: 'is-danger',
+            position: 'is-top',
+            indefinite: false,
+          })
         })
-        .catch(e => {})
+    },
+    addRule: function (event) {
+      this.parseSigma(this.editingRule.rule_yaml)
+      if (this.save_button_text === "Create") {
+        ApiClient.getSigmaRuleByText(rule_yaml)
+          .then(response => {
+            let SigmaRule = response.data.objects[0]
+            this.parsed = SigmaRule
+            this.data_type_present = (SigmaRule['search_query'].includes("data_type") || SigmaRule['search_query'].includes("source_name"))
+          })
+          .catch(e => {
+          })
+      }
+      if (this.save_button_text === "Update") {
+        // Only update the rule if the parsing was positive.
+        ApiClient.getSigmaRuleByText(this.editingRule.rule_yaml)
+          .then(response => {
+            let SigmaRule = response.data.objects[0]
+            this.parsed = SigmaRule
+            ApiClient.updateSigmaRule(this.editingRule.id, this.editingRule.rule_yaml)
+              .then(response => { })
+              .catch(e => {
+                console.error(e)
+              })
+            location.reload();
+            this.$buefy.notification.open({ message: 'Succesfully modified Sigma rule!', type: 'is-success' })
+            this.showEditModal = false
+            this.data_type_present = (SigmaRule['search_query'].includes("data_type") || SigmaRule['search_query'].includes("source_name"))
+          })
+          .catch(e => {
+          })
+      }
+      //
+    },
+    composeRule() {
+      this.showEditModal = true
+      this.create_new_rule = true
+      this.save_button_text = "Create"
+      this.editingRule = { rule_yaml: this.text }
+    },
+    startRuleEdit(rule) {
+      if (rule === null)
+        console.log("No rule given, maybe you wanna add one?")
+      this.save_button_text = "Update"
+      this.showEditModal = true
+      this.editingRule = rule
+    },
+    deleteRule(ioc) {
+      if (confirm('Delete Rule?')) {
+        ApiClient.deleteSigmaRule(ioc.rule_uuid)
+          .then(response => { })
+          .catch(e => {
+            console.error(e)
+          })
+        // TODO: remove the element from the array
+      }
+      // instead of removing the element from the array gonna relad the page
+      location.reload();
```

Comment:
Why? It's much more expensive and slow to reload the page than to remove the element from the array.

Reply:
fixed